twamm-ekubo/src/L1TWAMMBridge.sol

```diff
@@ -105,53 +146,139 @@ contract L1TWAMMBridge is Ownable {
+            msg.sender,
+            params.sellToken,
+            params.buyToken,
+            params.fee,
+            params.start,
+            params.end,
+            params.amount
+        );
+        starknetBridge.depositWithMessage{value: msg.value}(
+            address(token),
+            params.amount,
+            l2EndpointAddress,
+            payload
         );
-        _depositWithMessage(params.amount, payload, address(tokenBridge), msg.value);
 
-        emit DepositAndCreateOrder(msg.sender, l2EndpointAddress, params.amount, DEFAULT_NONCE);
+        emit DepositAndCreateOrder(
+            msg.sender,
+            l2EndpointAddress,
+            params.amount,
+            DEFAULT_NONCE
+        );
     }
 
     /// @notice Initiates a withdrawal from L2 to L1
-    /// @param amount Amount of tokens to withdraw
+    /// @param receiver Address of the receiver
+    /// @param order_id Amount of tokens to withdraw
     /// @param l1_token Address of the L1 token to receive
     /// @dev Requires msg.value to cover messaging fees
-    function initiateWithdrawal(uint256 amount, address l1_token) external payable {
-        uint256[] memory message = _encodeWithdrawalPayload(msg.sender, l1_token, amount);
+    function initiateWithdrawal(
+        address receiver,
+        address l1_token,
+        uint64 order_id
+    ) external payable {
+        uint256[] memory message = _encodeWithdrawalPayload(
+            msg.sender,
+            receiver,
+            l1_token,
+            order_id
+        );
 
-        _sendMessage(l2EndpointAddress, ON_RECEIVE_SELECTOR, message, msg.value);
+        _sendMessage(
+            l2EndpointAddress,
+            ON_RECEIVE_SELECTOR,
+            message,
+            msg.value
+        );
 
-        emit WithdrawalInitiated(msg.sender, amount);
+        emit WithdrawalInitiated(msg.sender, order_id);
+    }
+
+    function claimWithdrawal(
+        address _token,
+        uint256 _amount,
+        address _recipient
+    ) external{
+        address tokenBridge = starknetRegistry.getBridge(_token);
+        IStarknetTokenBridge(tokenBridge).withdraw(_token, _amount, _recipient);
     }
 
     /// @notice Initiates a request to cancel a deposit
-    /// @param l1_token Address of the token being deposited
-    /// @param amount Amount of tokens being deposited
+    /// @param l1_token Address of the token deposited
     /// @param nonce Unique identifier for the deposit
-    function initiateCancelDepositRequest(address l1_token, uint256 amount, uint256 nonce) external {
-        address tokenBridge = starknetRegistry.getBridge(address(token));
-        IStarknetTokenBridge(tokenBridge).depositCancelRequest(l1_token, amount, l2EndpointAddress, nonce);
+    function initiateCancelDepositRequest(
+        address l1_token,
+        OrderParams memory params,
+        uint256 nonce
+    ) external {
+        // address tokenBridge = starknetRegistry.getBridge(address(token));
```

Comment:
why is this commented out? If we don't need these remove them but my guess is we do, i could be wrong though